app/models/user.rb

```diff
@@ -1,8 +1,10 @@
 # Huginn is designed to be a multi-User system.  Users have many Agents (and Events created by those Agents).
 class User < ActiveRecord::Base
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable, :lockable,
-         :omniauthable
+  DEVISE_MODULES = [:database_authenticatable, :registerable,
+                    :recoverable, :rememberable, :trackable,
+                    :validatable, :lockable, :omniauthable,
+                    (ENV['ALLOW_UNCONFIRMED_ACCESS'] == 'false' ? :confirmable : nil)].compact
```

Comment:
Yup that sounds great to me.

On Thu, Mar 17, 2016 at 2:29 AM Dominik Sander <notifications@github.com>
wrote:

> In app/models/user.rb
> <https://github.com/cantino/huginn/pull/1330#discussion_r56475763>:
>
> > @@ -1,8 +1,10 @@
> >  # Huginn is designed to be a multi-User system.  Users have many Agents (and Events created by those Agents).
> >  class User < ActiveRecord::Base
> > -  devise :database_authenticatable, :registerable,
> > -         :recoverable, :rememberable, :trackable, :validatable, :lockable,
> > -         :omniauthable
> > +  DEVISE_MODULES = [:database_authenticatable, :registerable,
> > +                    :recoverable, :rememberable, :trackable,
> > +                    :validatable, :lockable, :omniauthable,
> > +                    (ENV['ALLOW_UNCONFIRMED_ACCESS'] == 'false' ? :confirmable : nil)].compact
>
> Very good point, if found that a little confusing myself, of course just
> inverting the logic did not came to mind [image: :smile:]. I went with
> REQUIRE_CONFIRMED_EMAIL though, does that work for you?
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/cantino/huginn/pull/1330/files/df349a73028f623ba9e0dbf68ac6736e1134f2a4#r56475763>
>